diff --git a/build.sbt b/build.sbt
index 2313f634f..a2c8241dd 100644
--- a/build.sbt
+++ b/build.sbt
@@ -62,17 +62,17 @@ lazy val chiselSettings = Seq(
 
 lazy val `api-config-chipsalliance` = (project in file("api-config-chipsalliance/build-rules/sbt"))
   .settings(commonSettings)
-  .settings(publishArtifact := false)
+  //.settings(publishArtifact := false)
 lazy val hardfloat  = (project in file("hardfloat"))
   .settings(commonSettings, chiselSettings)
-  .settings(publishArtifact := false)
+  //.settings(publishArtifact := false)
 lazy val `rocket-macros` = (project in file("macros")).settings(commonSettings)
-  .settings(publishArtifact := false)
+  //.settings(publishArtifact := false)
 lazy val rocketchip = (project in file("."))
   .settings(commonSettings, chipSettings, chiselSettings)
-  .dependsOn(`api-config-chipsalliance`)
-  .dependsOn(hardfloat)
-  .dependsOn(`rocket-macros`)
+  .dependsOn(`api-config-chipsalliance`).aggregate(`api-config-chipsalliance`)
+  .dependsOn(hardfloat).aggregate(hardfloat)
+  .dependsOn(`rocket-macros`).aggregate(`rocket-macros`)
   .settings( // Assembly settings
     assembly / test := {},
     assembly / assemblyJarName := "rocketchip.jar",
