diff --git a/build.sc b/build.sc
index cf602b2a8..cd7499f2c 100644
--- a/build.sc
+++ b/build.sc
@@ -25,6 +25,16 @@ object v {
   )
 }
 
+object GitPublishVersion {
+  def publishVersion(cross: Option[String], nominalPublishVersion: String): String = {
+      val versionSansSnapshot = nominalPublishVersion.split('-').head
+      val gitHashCmd = os.proc("git", "rev-parse", "--short", "HEAD").call()
+      val gitHash = gitHashCmd.out.text().trim()
+      val crossFragment = cross.map{ c => "-" + c }.getOrElse("")
+      versionSansSnapshot + crossFragment + "-" + gitHash + "-SNAPSHOT"
+  }
+}
+
 // Build form source only for dev
 object chisel extends Chisel
 
@@ -45,6 +55,8 @@ trait Macros
   def scalaVersion: T[String] = T(v.scala)
 
   def scalaReflectIvy = v.scalaReflect
+
+  override def publishVersion: T[String] = T(GitPublishVersion.publishVersion(None, super.publishVersion()))
 }
 
 object hardfloat extends mill.define.Cross[Hardfloat](v.chiselCrossVersions.keys.toSeq)
@@ -67,6 +79,8 @@ trait Hardfloat
   def chiselPluginIvy = Option.when(crossValue != "source")(v.chiselCrossVersions(crossValue)._2)
 
   def repositoriesTask = T.task(super.repositoriesTask() ++ v.sonatypesSnapshots)
+
+  override def publishVersion: T[String] = T(GitPublishVersion.publishVersion(Some(crossValue), super.publishVersion()))
 }
 
 object cde extends CDE
@@ -79,6 +93,8 @@ trait CDE
   def scalaVersion: T[String] = T(v.scala)
 
   override def millSourcePath = os.pwd / "dependencies" / "cde" / "cde"
+
+  override def publishVersion: T[String] = T(GitPublishVersion.publishVersion(None, super.publishVersion()))
 }
 
 object diplomacy extends mill.define.Cross[Diplomacy](v.chiselCrossVersions.keys.toSeq)
@@ -104,6 +120,8 @@ trait Diplomacy
   def cdeModule = cde
 
   def sourcecodeIvy = v.sourcecode
+
+  override def publishVersion: T[String] = T(GitPublishVersion.publishVersion(Some(crossValue), super.publishVersion()))
 }
 
 object rocketchip extends Cross[RocketChip](v.chiselCrossVersions.keys.toSeq)
@@ -140,13 +158,15 @@ trait RocketChip
   def json4sJacksonIvy = v.json4sJackson
 
   def repositoriesTask = T.task(super.repositoriesTask() ++ v.sonatypesSnapshots)
+
+  override def publishVersion: T[String] = T(GitPublishVersion.publishVersion(Some(crossValue), super.publishVersion()))
 }
 
 trait RocketChipPublishModule
   extends PublishModule {
   def pomSettings = PomSettings(
     description = artifactName(),
-    organization = "org.chipsalliance",
+    organization = "edu.berkeley.cs",
     url = "http://github.com/chipsalliance/rocket-chip",
     licenses = Seq(License.`Apache-2.0`),
     versionControl = VersionControl.github("chipsalliance", "rocket-chip"),
